### DISCUSSION TITLE
Python script to check metabolic tasks

### DISCUSSION THREAD

**Discussion by raquelgg11** [SysBioChalmers/Human-GEM#686] [General]
Let me introduce myself. I am Raquel Garcia, from the University the Murcia (Spain). My Master Thesis is about to check the metabolic tasks you proposed. This work has been done under the supervision of Profs. F. Guil and José M. García.

In my Master Thesis, we propose a methodology to search for genetic interventions or vulnerabilities, based on the analysis of essential genes and synthetic lethality gene pairs, which are indispensable for the execution of a set of 57 metabolic tasks crucial for cell viability. The fundamental purpose of this research is to broaden the range of possible therapeutic targets, both in the context of cancer cells and in the context of possible toxicities in healthy cells.

Following your proposal, we have developed several python scripts to verify the performance of the 57 metabolic tasks you described in https://www.science.org/doi/10.1126/scisignal.aaz1482, using Human 1.14 and other ftINIT models. 

We've taken your data set and applied the convenient bound constraints. We've created the necessary reactions and optimized them using COBRApy methods.

Therefore, our work generates 57 new models, one for each metabolic task. Then, these new models can be used to carry out analysis of essential genes or synthetic lethality for each task, broadening the variety of genetic interventions in human cells by the implementation of the Metabolic Tasks.

We consider that it will be useful to include our work in your GitHub. 

Finally, as an example, I will show you how we have checked the task “Aerobic rephosphorylation of ATP from glucose” 

```
import cobra
from cobra import Model, Reaction, Metabolite

model = cobra.io.read_sbml_model("human.xml")

with model:    
    interes = Reaction('AerobicRephosphorylationOfATP') 
    interes.lower_bound = 1.   
    interes.upper_bound = 1000.   
    interes.add_metabolites({ 

        model.metabolites.MAM01371c: -1.0, 
        model.metabolites.MAM02040c: -1.0, 
        model.metabolites.MAM01285c: 1, 
        model.metabolites.MAM02039c: 1, 
        model.metabolites.MAM02751c: 1 
    }) 

    model.add_reactions([interes]) 
    for r in model.exchanges: 
        r.lower_bound=0 
        r.upper_bound=0 

    O2.lower_bound=-1000 
    glucosa.lower_bound=-1000 
    H2O.upper_bound=1000 
    CO2.upper_bound=1000 
 
    sol=model.slim_optimize() 
    cobra.io.write_sbml_model(model,"Aerobic-rephosphorylation-of-ATP-from-glucose.xml") 
```

**Comment by mihai-sysbio** [comment]
Hi @raquelgg11! This sounds very nice - for a long time there have been discussions around the future directions of the metabolic tasks.

As you are probably aware, this repository only holds the generic model. All contextualized models are held elsewhere, e.g. Zenodo or other GitHub repositories.

To continue the discussion, I hope it's okay that I convert this issue into a GitHub Discussion such that an actionable solution can be found.

For the moment, my first questions are:
- to what extent is the work scripted? e.g., could it be run automatically?
- is there a vision to keep these scripts/model,s up-to-date?
- is there any documentation already available?

**Comment by haowang-bioinfo** [comment]
@raquelgg11 very interesting - look forward to your code
